clients/js/src/Collection.ts

```diff
@@ -0,0 +1,382 @@
+      this.client.api.options,
+    )) as MultiQueryResponse;
+  }
+
+  /**
+   * Modify the collection name or metadata
+   * @param {Object} params - The parameters for the query.
+   * @param {string} [params.name] - Optional new name for the collection.
+   * @param {CollectionMetadata} [params.metadata] - Optional new metadata for the collection.
+   * @returns {Promise<void>} - The response from the API.
+   *
+   * @example
+   * ```typescript
+   * const response = await client.updateCollection({
+   *   name: "new name",
+   *   metadata: { "key": "value" },
+   * });
+   * ```
+   */
+  async modify({
+    name,
+    metadata,
+  }: {
+    name?: string;
+    metadata?: CollectionMetadata;
+  }): Promise<CollectionParams> {
+    await this.client.init();
+    return this.client.api
+      .updateCollection(
+        this.id,
+        {
+          new_name: name,
+          new_metadata: metadata,
+        },
+        this.client.api.options,
+      )
+      .then(() => {
+        if (name !== undefined) {
+          this.name = name;
+        }
+        if (metadata !== undefined) {
+          this.metadata = metadata;
+        }
+        return {
+          name: this.name,
+          metadata: this.metadata,
+        } as CollectionParams;
+      });
+  }
+
+  /**
+   * Peek inside the collection
+   * @param {Object} params - The parameters for the query.
+   * @param {PositiveInteger} [params.limit] - Optional number of results to return (default is 10).
+   * @returns {Promise<GetResponse>} A promise that resolves to the query results.
+   * @throws {Error} If there is an issue executing the query.
+   *
+   * @example
+   * ```typescript
+   * const results = await collection.peek({
+   *   limit: 10
+   * });
+   * ```
+   */
+  async peek({ limit = 10 }: PeekParams = {}): Promise<MultiGetResponse> {
+    await this.client.init();
+    return (await this.client.api.aGet(
+      this.id,
+      {
+        limit,
+      },
+      this.client.api.options,
+    )) as MultiGetResponse;
+  }
+
+  /**
+   * Deletes items from the collection.
+   * @param {Object} params - The parameters for deleting items from the collection.
+   * @param {ID | IDs} [params.ids] - Optional ID or array of IDs of items to delete.
+   * @param {Where} [params.where] - Optional query condition to filter items to delete based on metadata values.
+   * @param {WhereDocument} [params.whereDocument] - Optional query condition to filter items to delete based on document content.
+   * @returns {Promise<string[]>} A promise that resolves to the IDs of the deleted items.
+   * @throws {Error} If there is an issue deleting items from the collection.
+   *
+   * @example
+   * ```typescript
+   * const results = await collection.delete({
+   *   ids: "some_id",
+   *   where: {"name": {"$eq": "John Doe"}},
+   *   whereDocument: {"$contains":"search_string"}
+   * });
+   * ```
+   */
+  async delete({ ids, where, whereDocument }: DeleteParams = {}): Promise<
```

Comment:
should we make DeleteParams non optional?

Reply:
If delete() results in noop, I think we can make DeleteParams non optional

Reply:
`.delete()` is a no-op